### DISCUSSION TITLE
HA Log Error: The preset_mode is not a valid preset_mode

### DISCUSSION THREAD

**Discussion by iHaveAstream** [SukramJ/aiohomematic#1452] [Q&A]
### The problem

Hi,
Since a while I get some error in the HA log as follows and each time I either enable another preset mode for heating or switch to Auto mode:

`Logger: homeassistant.components.websocket_api.http.connection
Source: components/websocket_api/commands.py:282
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 20:51:22 (1 occurrences)
Last logged: 20:51:22

[139881380757696] The preset_mode is not a valid preset_mode: boost, none, week_program_1, week_program_2, week_program_3, week_program_4, week_program_5, week_program_6`

The presets modes that I can select in HA are named differently, at least some of them... the week programs are named "WP 1", "WP 2", etc. But "Boost" and "None" do exist under same naming.

I know that this is not an error originated by this Integration, however probably someone has an idea to get rid of it.
Apparently, everything seems to work well, means I can control everything, even though that errror comes in the log.

Thank you!

**Comment by SukramJ** [comment]
>  enable another preset mode for heating or switch to Auto mode:

`Auto` is not a valid `preset_mode`. `Auto` is a `hvac_mode`

> valid preset_mode: boost, none, week_program_1, week_program_2, week_program_3, week_program_4, week_program_5, week_program_6`

Valid values for automations can be seen under `Dev tools --> states`

> WP 1", "WP 2", etc. But "Boost" and "None"

These are "translations" for the UI.

**Comment by iHaveAstream** [comment]
True, under Dev Tools > States I can see for example:

```
preset_modes:
  - boost
  - none
  - week_program_1
  - week_program_2
  - week_program_3
```

That mode `Auto` is not a `preset_mode` is clear - my mistake.

But, I'm not using any automations in HA to control my `hvac_mode` or `preset_mode`, everything is set directly in the CCU3.
So would there be a way to make to UI recognize the valid `preset_modes` & `hvac_mode` somehow to not generate log errors?

Thanks!

**Reply by SukramJ** [reply]
> `Logger: homeassistant.components.websocket_api.http.connection
Source: components/websocket_api/commands.py:282
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 20:51:22 (1 occurrences)
Last logged: 20:51:22

Looks like you are using the HA UI to switch the modes.

**Comment by iHaveAstream** [comment]
Yes, that's correct. Sometimes I do use the HA UI to switch modes.
But my point is rather if there's a way to "force" HA to use the correct namings for `preset_mode` and `hvac_mode` and not to use its "own" abbreviations like `WP 1` etc.
Interestingly, in the past there was no issue related to that and in HA UI the `preset_modes` were listed in short `WP 1` etc. ever since.

**Reply by SukramJ** [reply]
> in the past there ...

The past is a year ago. (< 1.26.0)

> But my point is rather if there's a way to "force" HA to use the correct namings for preset_mode and hvac_mode and not to use its "own" abbreviations like WP 1 etc.

The build in HA UI components and editors have no problem with translated values(WP 1) and send the correct value(week_program_1) to the backend.

So what is your problem? You need to fix it on your installation.

**Comment by iHaveAstream** [comment]
> The past is a year ago. (< 1.26.0)

time goes by fastly ^^

> So what is your problem? You need to fix it on your installation.

alright, would love to do so, but how? could you pls. give me a hint?

Much appreciate your support!
Thanks.

**Reply by SukramJ** [reply]
How can I give you a hint if I have to pull every detail out of you.

> Sometimes I do use the HA UI to switch modes.

Which component/card do you use? 
Used in UI / Automation /Script Scene ?

**Comment by iHaveAstream** [comment]
Actually I'm using cards only. For example:
![_2024-03-05_121320](https://github.com/danielperna84/hahomematic/assets/62019393/0ec552b8-ff7c-4544-a19d-386b5ebdc23b)

Interestingly, when I switch to YAML mode, the correct namings will be reflected:
![_2024-03-05_121606](https://github.com/danielperna84/hahomematic/assets/62019393/08637c03-18cb-4aa7-b5e2-8a0003ef5d4e)

**Reply by SukramJ** [reply]
This default card should not be a problem.

> Since a while I get some error in the HA log as follows and each time I either enable another preset mode for heating or switch to Auto mode

What are the step you use to switch to Auto mode?

**Comment by iHaveAstream** [comment]
> What are the step you use to switch to Auto mode?

just these in the red rectangle:
![_2024-03-05_121606](https://github.com/danielperna84/hahomematic/assets/62019393/cb02ee42-b9fa-42a4-8efe-d1d450821406)

so, I normally use the very right one ("Off") and then the very left one ("Auto").

**Reply by SukramJ** [reply]
OK. Looks like a frontend issue with added preset_mode to the card. Its sending an empty preset even though only the HVAC_Mode has changed.

**Reply by SukramJ** [reply]
Looks something like [that](https://github.com/home-assistant/frontend/issues/19620)

**Comment by iHaveAstream** [comment]
> Looks something like [that](https://github.com/home-assistant/frontend/issues/19620)

Indeed, looks kinda similar!
Also I (sometimes) get other "translation" errors in the log, related to other (HACS) integrations, such as Proxmox for instance.
I think this needs some investigation on HA frontend side, am I right?

**Reply by SukramJ** [reply]
I think so.

**Comment by iHaveAstream** [comment]
OK, got it! 
Since it seems rather to be something "cosmetic" than a functional problem, I can live with it.
Probably they'll fix it some day.
Thanks for your time and help - we can "close" this one here.
